### PR TITLE
Add module containing configuration info to reduce config errors

### PR DESCRIPTION
**Pull request by jahands**
This will fix the potential for errors when accessing configuration info.

It also makes it easier to use because you get intellisense for config keys rather than needing to copy/paste from `project.json`.

intellisense

### CODE REVIEW

File: test/extension.test.ts [hunk 1]
```diff
@@ -10,8 +10,9 @@ import * as assert from 'assert';
 // as well as import your extension to test it
 import * as vscode from 'vscode';
 import * as myExtension from '../src/extension';
-
+import * as cInfo from '../src/configinfo'
```
Comment (SrTobi) [inline test/extension.test.ts]: Missing ;

File: test/extension.test.ts [hunk 2]
```diff
@@ -19,4 +20,18 @@ suite("Extension Tests", () => {
...
-});
\ No newline at end of file
+});
+*/
+suite("Configuration Tests:", () => {
+	let config = vscode.workspace.getConfiguration(cInfo.group);
+	test("configInfo.keys only contains valid config keys", function() {
+		for (let key in cInfo.keys) {
+			if (cInfo.keys.hasOwnProperty(key)) {
+				assert(config.has(key), "Invalid configInfo.key: " + key)
```
Comment (SrTobi) [inline test/extension.test.ts]: missing ;

**Comment by SrTobi** [comment]
:+1:

**Comment by jahands** [comment]
Thanks for catching that! Merging.

**Pull request merged**